support/src/test/java/com/shopify/graphql/support/IntegrationTest.java

```diff
@@ -166,10 +167,26 @@ public void testOptionalFieldOnInput() throws Exception {
     }
 
     @Test
-    public void testUnsetOptionalFieldOnInput() throws Exception {
+    public void testOptionalFieldOnInputAsUndefined() throws Exception {
         String queryString = Generated.mutation(mutation -> mutation
-            .setInteger(new Generated.SetIntegerInput("answer", 42).setTtl(null).unsetTtl())
+          .setInteger(new Generated.SetIntegerInput("answer", 42).setTtlInput(Input.<LocalDateTime>undefined()))
         ).toString();
         assertEquals("mutation{set_integer(input:{key:\"answer\",value:42})}", queryString);
     }
+
+    @Test
+    public void testOptionalFieldOnInputAsExplicitNull() throws Exception {
+        String queryString = Generated.mutation(mutation -> mutation
+          .setInteger(new Generated.SetIntegerInput("answer", 42).setTtlInput(Input.<LocalDateTime>value(null)))
```

Comment:
It tests different way to set `null` as explicit input value to be sent to the server

Reply:
What I don't understand is why we are adding another way to do this when we already have a way.

Reply:
You mean why do we have another setter like `setTtlInput(Input<LocalDateTime>)` ?

To be able to do this: `setTtlInput(Input.undefined())` that means `ttl` is undefined don't send it to the server at all (don't send `null`). 

Why do we have 2 of them? Because of MBSDK depends on this and if we remove existing one it means breaking changes. So we keep 2 of them for now.